### PR TITLE
The Seed Extractor now vends into hands if possible

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

When vending seeds from botany's machine seed extractor, it will now try to place the chosen seeds into the user's hand or offhand first, before dropping it onto the ground.

Also removes a weird commented out to_chat message and gives it some more descriptive variables.

To clarify: Extracting seeds from a plant is unchanged. The only thing this adjusts is vending seeds from the machine.

## Why It's Good For The Game

Just like how the seed vendor drops seeds into your hand automatically, so does the seed extractor now, for consistency. 

## Changelog
:cl: Melbert
tweak: The seed extractor will now dispense seeds directly into hands first like vending machines.
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/seed_extractor.dm [hunk 1]
```diff
@@ -190,11 +190,18 @@
...
+						var/mob/user = usr
+						if(user.put_in_hands(found_seed))
+							to_chat(user, "<span class='notice'>[found_seed] drops into your hand.</span>")
+						else
+							to_chat(user, "<span class='notice'>[found_seed] drops onto the seed extractor.</span>")
+					else
+						found_seed.forceMove(drop_location())
+						visible_message("<span class='notice'>[found_seed] drops onto the floor.</span>", null, "<span class='hear'>You hear a soft clatter.</span>", COMBAT_MESSAGE_RANGE)
 					. = TRUE
-					//to_chat(usr, "<span class='notice'>[src] clanks to life briefly before vending [prize.equipment_name]!</span>")
```
Comment (MrMelbert) [inline code/modules/hydroponics/seed_extractor.dm:199]: It looks like someone intended to add a message to the seed extractor at some point, pasted it, then commented it out and left it there.

**Comment by Tlaltecuhtli** [comment]
why not insert in the machine?

**Comment by MrMelbert** [comment]
> 
> 
> why not insert in the machine?

this pr doesn't change the act of extracting seeds from plants, or the act of placing seeds into the machine
this only affect withdrawing seeds from the machine itself

File: code/modules/hydroponics/seed_extractor.dm [hunk 2]
```diff
@@ -190,11 +190,18 @@
...
 			var/item = params["item"]
 			if(piles[item] && length(piles[item]) > 0)
-				var/datum/weakref/WO = piles[item][1]
-				var/obj/item/seeds/O = WO.resolve()
-				if(O)
-					piles[item] -= WO
-					O.forceMove(drop_location())
+				var/datum/weakref/found_seed_weakref = piles[item][1]
+				var/obj/item/seeds/found_seed = found_seed_weakref.resolve()
+				if(found_seed)
```
Comment (Jared-Fogle) [inline code/modules/hydroponics/seed_extractor.dm:195]: Early return.

File: code/modules/hydroponics/seed_extractor.dm [hunk 3]
```diff
@@ -190,11 +190,18 @@
...
+					piles[item] -= found_seed_weakref
+					if(usr)
+						var/mob/user = usr
+						if(user.put_in_hands(found_seed))
+							to_chat(user, "<span class='notice'>You take \the [found_seed] out of the slot.</span>")
```
Comment (Jared-Fogle) [inline code/modules/hydroponics/seed_extractor.dm:200]:
```suggestion
							to_chat(user, "<span class='notice'>You take [found_seed] out of the slot.</span>")
```

Comment (Jared-Fogle) [reply code/modules/hydroponics/seed_extractor.dm:200]: The is automatic.

File: code/modules/hydroponics/seed_extractor.dm [hunk 4]
```diff
@@ -190,11 +190,18 @@
...
+						else
+							to_chat(user, "<span class='notice'>\The [found_seed] falls onto the floor.</span>")
```
Comment (Jared-Fogle) [inline code/modules/hydroponics/seed_extractor.dm:202]:
```suggestion
							to_chat(user, "<span class='notice'>[found_seed] falls onto the floor.</span>")
```

**Pull request merged**